lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -435,10 +435,14 @@ def check_watched_item(self, job_state):
 
             # This assumes jobs dependent on a single pod, single container
             if succeeded > 0:
-                self.__produce_log_file(job_state)
-                error_file = open(job_state.error_file, 'w')
-                error_file.write("")
-                error_file.close()
+                try:
+                    self.__produce_log_file(job_state)
+                    error_file = open(job_state.error_file, 'w')
+                    error_file.write("")
+                    error_file.close()
+                except IOError as e:
```

Comment:
Do you know where exactly the IOError is being raised ? I'd assume it's in `self.__produce_log_file(job_state)` ?
Or is it because the job directory doesn't exist anymore ?
In either case you can use 
```
with open(ob_state.error_file, 'w'):
    pass
```

Reply:
It was `self.__produce_log_file(job_state)`:
```
galaxy.jobs.runners ERROR 2018-09-28 16:54:07,723 Unhandled exception checking active jobs
Traceback (most recent call last):
  File "/export/galaxy-central/lib/galaxy/jobs/runners/__init__.py", line 594, in monitor
    self.check_watched_items()
  File "/export/galaxy-central/lib/galaxy/jobs/runners/__init__.py", line 621, in check_watched_items
    new_async_job_state = self.check_watched_item(async_job_state)
  File "/export/galaxy-central/lib/galaxy/jobs/runners/kubernetes.py", line 430, in check_watched_item
    self.__produce_log_file(job_state)
  File "/export/galaxy-central/lib/galaxy/jobs/runners/kubernetes.py", line 540, in __produce_log_file
    logs_file = open(logs_file_path, mode="w")
IOError: [Errno 2] No such file or directory: '/export/galaxy-central/database/job_working_directory/011/11266/galaxy_11266.o'
```

I presume because the job directory doesn't exist anymore.

Regarding your suggestion to use `with open...`, good call:  I'll patch the old code.